File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var getMinimumDifference = function(root) {
    let node = root;
    let vals = [];
    const traverse = (root) =>{
        vals.push(root.val);
        if(root.left) traverse(root.left);
        if(root.right) traverse(root.right);
    }
    traverse(node);
    vals.sort((a,b) => a - b)
    let min = Infinity;
    for(let i = 1; i < vals.length; i++){
        if((vals[i] - vals[i-1]) < min){
            min = vals[i] - vals[i-1]
        }
    }
    return min
};

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private Integer prev = null;
    private int minDiff = Integer.MAX_VALUE;
    
    public int getMinimumDifference(TreeNode root) {
        inorder(root);
        return minDiff;
    }
    
    private void inorder(TreeNode node) {
        if (node == null) return;

        inorder(node.left);
        if (prev != null) {
            minDiff = Math.min(minDiff, node.val - prev);
        }
        prev = node.val;
        inorder(node.right);
    }
}